pkg/operator/starter.go

```diff
@@ -143,22 +152,39 @@ type controllerTemplateData struct {
 // withCustomCABundle executes the asset as a template to fill out the parts required when using a custom CA bundle.
 // The `caBundleConfigMap` parameter specifies the name of the ConfigMap containing the custom CA bundle. If the
 // argument supplied is empty, then no custom CA bundle will be used.
-func withCustomCABundle(assetFunc func(string) []byte, kubeClient kubeclient.Interface) func(string) []byte {
-	templateData := controllerTemplateData{}
-	switch used, err := isCustomCABundleUsed(kubeClient); {
-	case err != nil:
-		klog.Fatalf("could not determine if a custom CA bundle is in use: %v", err)
-	case used:
-		templateData.CABundleConfigMap = cloudConfigName
-	}
-	return func(name string) []byte {
-		asset := assetFunc(name)
-		template := template.Must(template.New("template").Parse(string(asset)))
-		buf := &bytes.Buffer{}
-		if err := template.Execute(buf, templateData); err != nil {
-			klog.Fatalf("Failed to execute ")
+func withCustomCABundle(cloudConfigLister corev1listers.ConfigMapNamespaceLister) csidrivercontrollerservicecontroller.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		switch used, err := isCustomCABundleUsed(cloudConfigLister); {
+		case err != nil:
+			return fmt.Errorf("could not determine if a custom CA bundle is in use: %w", err)
```

Comment:
Just to confirm: returning an error here will degrade the operator. Is that what we want?

Reply:
Yes, in my opinion that is what we want. But if you think otherwise, it won't hurt my feelings. Since we are using a lister, an error here is either (1) that the informer has been closed or (2) that there are RBAC restrictions preventing access to the ConfigMap. Both of those seem to me to warrant the operator being marked as degraded.